src/disk_buffer_pool.cpp

```diff
@@ -67,17 +67,12 @@ namespace libtorrent
 	namespace {
 
 	// this is posted to the network thread
-	void watermark_callback(std::vector<boost::weak_ptr<disk_observer> >* cbs)
+	void watermark_callback(std::vector<boost::weak_ptr<disk_observer>> cbs)
```

Comment:
are you sure the vector will be moved into here? presumably boost::bind() allow its function object to be called more than once, in which case it cannot move any of its state in. perhaps I'm missing something